### PR TITLE
Added implementation for projects updating, PUT and PATCH methods

### PR DESCRIPTION
**Pull request by AlexanderSychev2005**
Developed robust API endpoints to handle PUT and PATCH requests for updating project information.
Changes in serializer, views.
PUT - update all field
PATH - Update an existing project without all fields required
{{base_url}}api/projects/1/
base_url - http://127.0.0.1:8000/
{
"project_name": "Example Project2",
"description": "Example description",
"industries": "education",
"goals": "goals",
"budget_needed": 100,
"budget_ready": 100,
"industry": "education",
"status": "approved"
}

### CODE REVIEW

**Review by OlesiaBarannik — APPROVED** [review]

File: startups/views.py [hunk 1]
```diff
@@ -207,37 +207,57 @@ def create(self, request):
...
-    #     project_specific_updated_info = request.data
-    #     data = {
-    #         'project_id': project_id,
-    #         'message': f"Hello, here's a PATCH method! You update SOME information about project № {project_id}",
-    #         'specific_updated_data': project_specific_updated_info,
-    #         'status': status.HTTP_200_OK
-    #     }
-    #     return Response(data, status=status.HTTP_200_OK)
+
+    def update(self, request, pk=None):
```
Comment (VovaD199) [inline startups/views.py:211]: Implemented an interesting solution for checking the presence of "pk", which entails redundant checks. I recommend removing the "pk=None" parameter, and in the "try" block, try to extract it from the request, if not successful, then process the error by displaying the appropriate notification, if successful, then update the data.

File: startups/views.py [hunk 2]
```diff
@@ -207,37 +207,57 @@ def create(self, request):
...
+        data = {
+            'project_id': pk,
+            'message': f"Hello, here's a PUT method! You update ALL information about PROJECT № {pk}",
+            'updated_data': request.data,
+            'status': status.HTTP_200_OK
+        }
+        return Response(data, status=status.HTTP_200_OK)
+
+
+    def partial_update(self, request, pk=None):
```
Comment (VovaD199) [inline startups/views.py:237]: Implemented an interesting solution for checking the presence of "pk", which entails redundant checks. I recommend removing the "pk=None" parameter, and in the "try" block, try to extract it from the request, if not successful, then process the error by displaying the appropriate notification, if successful, then update the data.

File: startups/views.py (hunk 1 above)
Comment (AlexanderSychev2005) [reply startups/views.py]: I see what you mean. I've removed None from args and moved checking into try except block to verify that the project exists and to verify the pk argument is valid.

File: startups/views.py (hunk 2 above)
Comment (AlexanderSychev2005) [reply startups/views.py]: I see what you mean. I've removed None from args and moved checking into try except block to verify that the project exists and to verify the pk argument is valid.

**Review by VovaD199 — APPROVED** [review]

**Pull request merged**